From: Simon McVittie <smcv@debian.org>
Date: Sun, 31 Dec 2017 00:16:19 +0000
Subject: Skip known-failing tests on 32-bit architectures at build-time

Failure to build from source is a release-critical bug, but these
misrenderings really don't look release-critical.

Bug: https://gitlab.gnome.org/GNOME/librsvg/issues/178
Bug-Debian: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=884849
Bug-Debian: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=884850
Forwarded: no, Debian-specific triaging
---
 tests/rsvg-test.c | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/tests/rsvg-test.c b/tests/rsvg-test.c
index 03a58cb..35c9aac 100644
--- a/tests/rsvg-test.c
+++ b/tests/rsvg-test.c
@@ -257,6 +257,17 @@ rsvg_cairo_check (gconstpointer data)
     if (g_str_has_suffix (test_file_base, ".svg"))
       test_file_base[strlen (test_file_base) - strlen (".svg")] = '\0';
 
+    if ((g_str_has_suffix (test_file_base, "/masking-mask-01-b") ||
+         g_str_has_suffix (test_file_base, "/masking-path-04-b")) &&
+        sizeof (void *) == 4 &&
+        g_getenv ("DEB_BUILD_TIME_TESTS") != NULL)
+    {
+        g_test_skip ("Known to fail on 32-bit architectures, "
+                     "https://gitlab.gnome.org/GNOME/librsvg/issues/178");
+        g_free (test_file_base);
+        return;
+    }
+
     rsvg = rsvg_handle_new_from_gfile_sync (test_file, 0, NULL, &error);
     g_assert_no_error (error);
     g_assert (rsvg != NULL);
